addons/stock/models/stock_move.py

```diff
@@ -413,24 +412,19 @@ def write(self, vals):
             #propagate also any delta observed when setting the move as done
             propagated_date_field = 'date'
 
-        if not self._context.get('do_not_propagate', False) and (propagated_date_field or propagated_changes_dict):
+        if propagated_date_field:
             #any propagation is (maybe) needed
             for move in self:
                 if move.move_dest_ids and move.propagate:
-                    if 'date_expected' in propagated_changes_dict:
-                        propagated_changes_dict.pop('date_expected')
```

Comment:
you can remove "if propagated_date_field:" because it is the only condition to get there